integrations/clickup/client.py

```diff
@@ -0,0 +1,230 @@
+import asyncio
+import time
+from typing import Any, AsyncGenerator, Optional, List, Dict
+from httpx import HTTPStatusError, Timeout
+from loguru import logger
+from port_ocean.utils import http_async_client
+from port_ocean.utils.cache import cache_iterator_result
+
+TEAM_OBJ = "__team"
+WEBHOOK_EVENTS = [
+    "taskCreated",
+    "taskUpdated",
+    "taskDeleted",
+    "listCreated",
+    "listUpdated",
+    "listDeleted",
+]
+MINIMUM_LIMIT_REMAINING = 20
+DEFAULT_SLEEP_TIME = 30
```

Comment:
May I know what influenced the choice for these numbers? Perhaps you can share some links?

Reply:
MINIMUM_LIMIT_REMAINING (20): This value ensures that the client starts waiting before hitting the exact rate limit, providing a buffer to avoid breaching it.
DEFAULT_SLEEP_TIME (30 seconds): This is the default wait time if the rate limit is hit, ensuring the client waits long enough for the rate limit to reset.